### PR TITLE
check user provided storage classes on the destination cluster

### PR DESCRIPTION
**Pull request by alaypatel07**
addresses #69 on destination cluster

### CODE REVIEW

**Review by pranavgaikwad — APPROVED** [review]
LGTM

File: 2_pvc_destination_gen/vars/storage-class-mappings.yml.example [hunk 1]
```diff
@@ -1,4 +1,4 @@
 mig_storage_class_mappings:
   glusterfs-storage_RWO: ocs-storagecluster-ceph-rbd
   glusterfs-storage_RWX: ocs-storagecluster-cephfs
-  glusterfs-storage-block_RWO: ocs_storagecluster-ceph-rbd
\ No newline at end of file
+  glusterfs-storage-block_RWO: ocs-storagecluster-ceph-rbd
```
Comment (djwhatle) [inline 2_pvc_destination_gen/vars/storage-class-mappings.yml.example:4]: nice catch

**Comment by djwhatle** [comment]
@alaypatel07 how do we handle PVCs without a designated storageClass? E.g. I think NFS PVs are storage class free. Maybe this isn't a concern given the narrow scope of OCS based environment we want this to run in.

File: 2_pvc_destination_gen/pvc-destination-gen.yml [hunk 2]
```diff
@@ -35,19 +35,38 @@
...
+        - fail:
+          when: "{{ (user_storage_classes | difference(available_storage_classes)) | length > 0 }}"
 
         - set_fact:
             pvc_data: "{{ lookup('file', pvc_data_filepath) | from_json }}"
             namespace_data: "{{ lookup('file', namespace_data_filepath) | from_json }}"
 
 
-    - name: ensure the namespaces are created for the pvcs
+    - name: "Ensure the namespaces are created for the pvcs"
```
Comment (djwhatle) [inline 2_pvc_destination_gen/pvc-destination-gen.yml:63]: +1

File: 2_pvc_destination_gen/pvc-destination-gen.yml [hunk 3]
```diff
@@ -35,19 +35,38 @@
         - fail:
           when: n.stat.exists == False
```
Comment (pranavgaikwad) [inline 2_pvc_destination_gen/pvc-destination-gen.yml]: @alaypatel07 This is unrelated to your PR. But while you're at it, can you please change `n.stat.exists == False` to `not n.stat.exists`

File: 2_pvc_destination_gen/vars/storage-class-mappings.yml.example (hunk 1 above)
Comment (alaypatel07) [reply 2_pvc_destination_gen/vars/storage-class-mappings.yml.example:4]: the script caught it, kept complaining that SC does not exist :P

Comment (djwhatle) [reply 2_pvc_destination_gen/vars/storage-class-mappings.yml.example:4]: i wondered :)

**Review by pranavgaikwad — APPROVED** [review]

**Review by djwhatle — APPROVED** [review]
